app/services/course/assessment/programming_evaluation_service.rb

```diff
@@ -107,7 +108,7 @@ def execute
   def create_evaluation
     Course::Assessment::ProgrammingEvaluation.create(
       course: @course, language: @language, package_path: @package, memory_limit: @memory_limit,
-      time_limit: @time_limit.to_i
+      time_limit: @time_limit
```

Comment:
Yes it automagically converts to Fixnum when creating `Course::Assessment::ProgrammingEvaluation`. It's probably the same behaviour as described in http://stackoverflow.com/questions/1329967/tell-ruby-program-to-wait-some-amount-of-time